cmd/collectors/restperf/plugins/volumetopclients/volumetopclients.go

```diff
@@ -0,0 +1,409 @@
+		return err
+	}
+
+	if err := v.processTopClientsByMetric(readDataVolumes, readDataSvms, topClientReadDataMatrix, "throughput.read", v.setDataMetric); err != nil {
+		return err
+	}
+
+	if err := v.processTopClientsByMetric(writeDataVolumes, writeDataSvms, topClientWriteDataMatrix, "throughput.write", v.setDataMetric); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (v *Volume) collectMetricValues(data *matrix.Matrix, filteredDataInstances *set.Set) ([]MetricValue, []MetricValue, []MetricValue, []MetricValue) {
+	readOpsMetric := data.GetMetric("total_read_ops")
+	writeOpsMetric := data.GetMetric("total_write_ops")
+	readDataMetric := data.GetMetric("bytes_read")
+	writeDataMetric := data.GetMetric("bytes_written")
+
+	var readOpsList, writeOpsList, readDataList, writeDataList []MetricValue
+	for _, key := range filteredDataInstances.Slice() {
+		var readOps, writeOps, readData, writeData float64
+		var ro, wo, rb, wb bool
+		if readOpsMetric != nil {
+			readOps, ro = readOpsMetric.GetValueFloat64(data.GetInstance(key))
+		}
+		if writeOpsMetric != nil {
+			writeOps, wo = writeOpsMetric.GetValueFloat64(data.GetInstance(key))
+		}
+		if ro && readOps > 0 {
+			readOpsList = append(readOpsList, MetricValue{key: key, value: readOps})
+		}
+		if wo && writeOps > 0 {
+			writeOpsList = append(writeOpsList, MetricValue{key: key, value: writeOps})
+		}
+
+		if readDataMetric != nil {
+			readData, rb = readDataMetric.GetValueFloat64(data.GetInstance(key))
+		}
+		if writeDataMetric != nil {
+			writeData, wb = writeDataMetric.GetValueFloat64(data.GetInstance(key))
+		}
+
+		if rb && readData > 0 {
+			readDataList = append(readDataList, MetricValue{key: key, value: readData})
+		}
+		if wb && writeData > 0 {
+			writeDataList = append(writeDataList, MetricValue{key: key, value: writeData})
+		}
+	}
+	return readOpsList, writeOpsList, readDataList, writeDataList
+}
+
+func (v *Volume) getTopN(metricList []MetricValue) []MetricValue {
+	sort.Slice(metricList, func(i, j int) bool {
+		return metricList[i].value > metricList[j].value
+	})
+
+	if len(metricList) > v.maxVolumeCount {
+		return metricList[:v.maxVolumeCount]
+	}
+	return metricList
+}
+
+func (v *Volume) extractVolumesAndSvms(data *matrix.Matrix, topMetrics []MetricValue) (*set.Set, *set.Set) {
+	volumes := set.New()
+	svms := set.New()
+	for _, item := range topMetrics {
+		instance := data.GetInstance(item.key)
+		if instance != nil {
+			volName := instance.GetLabel("volume")
+			svmName := instance.GetLabel("svm")
+			volumes.Add(volName)
+			svms.Add(svmName)
+		}
+	}
+	return volumes, svms
+}
+
+func (v *Volume) processTopClientsByMetric(volumes, svms *set.Set, matrixName, metric string, setMetricFunc func(*matrix.Matrix, *matrix.Instance, float64)) error {
+	if svms.Size() == 0 || volumes.Size() == 0 {
+		return nil
+	}
+
+	topClients, err := v.fetchTopClients(volumes.Values(), svms.Values(), metric)
```

Comment:
To reduce unnecessary allocations it would be better to pass the set and iterate over it instead of creating a copy with Values()